### ISSUE TITLE
Using DIsMax loss for my own classifier.

### ISSUE THREAD

**Issue by sparshgarg23** [dlmacedo/distinction-maximization-loss#1]
Hi,
Thanks for open sourcing your work.
Let's say my current classifer is  a 3 layer MLP with Dropout as shown below
'''
model.classifier=nn.Sequential(
nn.Linear(in_features=1536,out_features=625),
nn.ReLU(),
nn.Dropout(p=0.3),
nn.Linear(in_features=625,out_features=256)
nn.Linear(in_features=256,out_features=8),
)
'''
As such what changes shall i make in this in order to use  the distinction maximzation loss
From the readme model must be defined as follows
```
class Model(nn.Module):
    def __init__(self):
    (...)
    #self.classifier = nn.Linear(num_features, num_classes)
    self.classifier = losses.DisMaxLossFirstPart(num_features, num_classes)
```

**Comment by dlmacedo** [comment]
Hi,

I guess you can try:

`model.classifier[4] = losses.DisMaxLossFirstPart(256, 8)`

Do not forget the other changes you need to do!

David

**Comment by sparshgarg23** [comment]
Thanks for replying back,if you don't mind why did you mention  model.classifier[4]? and why not simply model.classifier?

**Comment by dlmacedo** [comment]
Because the classifier, from a DisMax point of view, is just the very last linear layer.

model.classifier in your code is actually a neural network. You may call all the layers a "deep" classifier.

However, from DisMax point of view, just the very last linear layer is considered the actual classifier.

**Comment by dlmacedo** [comment]
If you do:

`model.classifier = losses.DisMaxLossFirstPart(256, 8)
`

Then you will lose all layers! Not just the last one!

DisMax is supposed to replace just the very last linear layer.

**Comment by dlmacedo** [comment]
Important!

Rather than this:

```
#criterion = nn.CrossEntropyLoss()
criterion = losses.DisMaxLossSecondPart(model.classifier)
```
You will probably need to do this:

```
#criterion = nn.CrossEntropyLoss()
criterion = losses.DisMaxLossSecondPart(model.classifier[4])
```

**Comment by sparshgarg23** [comment]
That makes sense, thanks.
Also is the repository available to be used for both commercial as well as non-commercial purposes. If it's no for the commercial purpose, that is understandable.

I would still like to experiment with it for non -commercial purposes if it's okay with you
Thanks

**Comment by dlmacedo** [comment]
The code is available for non-commercial purposes.
